ext/standard/url.c

```diff
@@ -307,6 +307,8 @@ PHPAPI php_url *php_url_parse_ex(char const *str, size_t length)
 		if (p < e) {
 			ret->query = zend_string_init(p, (e - p), 0);
 			php_replace_controlchars_ex(ZSTR_VAL(ret->query), ZSTR_LEN(ret->query));
+		} else {
+			ret->query = ZSTR_EMPTY_ALLOC();
```

Comment:
Shouldn't the same be done for the `#` case above?

Reply:
thought, it's better to do it in another PR
but I could add it for current, if you think it's better

Reply:
I'd prefer to do both in the same PR, to make sure that the behavior between these two cases stays consistent.